File: src/Business/Services/CategoryGroupService.java
package Business.Services;

import Business.Services.Base.*;
import Common.DataAccess.*;
import Common.Models.*;
import Common.Services.*;

public class CategoryGroupService extends ReferenceParentEntityService<CategoryGroup, Category> implements ICategoryGroupService {
    public CategoryGroupService(
            IGlobalDataAccess globalDataAccess,
            ICategoryGroupDataAccess entityDataAccess,
            ICategoryDataAccess childEntityDataAccess) {
        super(globalDataAccess, entityDataAccess, childEntityDataAccess);
    }
}

File: backend/Business.Impl/Services/CategoryGroupService.cs
using Business.Contracts.Services;
using Business.Impl.Services.Base;
using Business.Models.Entities;
using DataAccess.Contracts;
using Shared.Contracts;

namespace Business.Impl.Services;

internal sealed class CategoryGroupService : GroupService<CategoryGroup, Category>, ICategoryGroupService
{
	public CategoryGroupService(ISharedContext sharedContext, IAppUnitOfWork unitOfWork)
		: base(sharedContext, unitOfWork, unitOfWork.CategoryGroupRepo)
	{
	}
}
